app/src/main/java/net/gsantner/markor/ui/hleditor/HighlightingEditor.java

```diff
@@ -156,10 +155,17 @@ public void sendAccessibilityEventUnchecked(AccessibilityEvent event) {
         }
     }
 
+    public void setBeforeHighlightCallback(final Callback.a0 callback) {
+        _beforeHighlightCallback = callback;
+    }
+
     private void highlightWithoutChange() {
-        final Editable editable = getText();
-        final boolean isFileShortEnough = editable.length() <= (_isDeviceGoodHardware ? 100000 : 35000);
-        if (isFileShortEnough && _hlEnabled) {
+        if (_beforeHighlightCallback != null) {
```

Comment:
@gsantner What do you think of this approach? This checks the file size on every highlight and disables the highlighting when size > threshold

Another, simpler way is to just check on file load.

Reply:
I think overall it's more complicated than it should be.  
Already thought about it and have something in mind, but not sure when I can make it.

What I would go for:
* Highlighter class(es) 
  * No limit in the  enable/disable supplied from outside
* DocumentActivity
  * `AppSettings::highlightingEnabled(File file, int filesize/characters/whatever)`
  * return first user preference set
  * then check limit and set default false+return in case too big
  * otherwise return global value
 
This not covers the case where a file would grow XXL size, but assuming you already had a file once open it doesn't change that much - and even if so, you can always disable it still then. The MOST important case is "first time open of a huge file".